internal/controller/utils.go

```diff
@@ -0,0 +1,147 @@
+package controller
+
+import (
+	"fmt"
+	"github.com/kintohub/kinto-cli/internal/api"
+	"github.com/kintohub/kinto-cli/internal/config"
+	"github.com/kintohub/kinto-cli/internal/types"
+	"github.com/kintohub/kinto-cli/internal/utils"
+)
+
+func (c *Controller) GetEnvFromId(envId string) *types.ClusterEnvironment {
+
+	envs, err := c.api.GetClusterEnvironments()
+	if err != nil {
+		utils.TerminateWithError(err)
+	}
+
+	for _, env := range envs {
+
+		if env.Id == envId {
+			return env
+		}
+	}
+	return nil
+}
+
+//Get All teleport-able/Accessible Environment names
+func (c *Controller) GetAvailableEnvNames(enableLocalGitCheck bool) []api.EnvDetails {
+
+	if enableLocalGitCheck {
+		utils.CheckLocalGitOrDie()
+	}
+
+	var envDetails []api.EnvDetails
+	serialNumber := 1
+
+	envs, err := c.api.GetClusterEnvironments()
+	if err != nil {
+		utils.TerminateWithError(err)
+	}
+
+	for _, env := range envs {
+		blocks, err := c.api.GetBlocks(env.Id)
+		if err != nil {
+			utils.TerminateWithError(err)
+		}
+
+		for _, block := range blocks {
+			latestRelease := utils.GetLatestSuccessfulRelease(block.Releases)
+
+			if latestRelease != nil && enableLocalGitCheck &&
+				utils.CompareGitUrl(latestRelease.BuildConfig.Repository.Url) {
+
+				envDetails = append(envDetails, api.EnvDetails{EnvName: fmt.Sprintf(
+					"%d. %s", serialNumber, env.Name), EnvId: env.Id, ClusterId: env.ClusterId})
+				serialNumber++
```

Comment:
u can remove the 

```
serialNumber++
```

from both if/else and move it at the end of the for loop after both of them

Reply:
why not rename it to `count` too? 
why is it called serial number?

Reply:
`count` and `serialNumber` are different. `count` keeps track of the port incrementation wherever necssary while `serialNumber` is used to avoid duplication of env/svs names when showing in the UI. this is due to the nature of the filteration process used to get the required env/svs.